From 5c2497b5c2468d6b79f960f547e61cec69c93adc Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Sun, 6 Mar 2016 08:54:19 +0300
Subject: 20_linux_xen: fix test for GRUB_DEVICE

Same fix as in 082bc9f.

Origin: upstream, http://git.savannah.gnu.org/cgit/grub.git/commit/?id=9545a8fd04ca33bcecc1907fec4d2354b02992ba
Bug-Debian: https://bugs.debian.org/841680
Bug-Debian: https://bugs.debian.org/841741
Last-Update: 2016-10-31

Patch-Name: uuid-test-precedence-2.patch
---
 util/grub.d/20_linux_xen.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index 9c4901fd7..6fd17f1e9 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -53,7 +53,7 @@ esac
 # and mounting btrfs requires user space scanning, so force UUID in this case.
 if [ "x${GRUB_DEVICE_UUID}" = "x" ] || [ "x${GRUB_DISABLE_LINUX_UUID}" = "xtrue" ] \
     || ! test -e "/dev/disk/by-uuid/${GRUB_DEVICE_UUID}" \
-    || test -e "${GRUB_DEVICE}" && uses_abstraction "${GRUB_DEVICE}" lvm; then
+    || ( test -e "${GRUB_DEVICE}" && uses_abstraction "${GRUB_DEVICE}" lvm ); then
   LINUX_ROOT_DEVICE=${GRUB_DEVICE}
 else
   LINUX_ROOT_DEVICE=UUID=${GRUB_DEVICE_UUID}
